src/qcodes/instrument_drivers/cryomagnetics/_cryomagnetics4g.py

```diff
@@ -298,17 +301,18 @@ def set_field(self, field_setpoint: float, block: bool = True) -> None:
             self.log.debug("Finished blocking ramp")
             # If we are now holding, it was successful
 
-            if not exit_state.holding:
-                msg = "_set_field({}) failed with state: {}"
-                raise Cryomagnetics4GException(msg.format(field_setpoint, exit_state))
+            if not exit_state.standby:
```

Comment:
```suggestion
            if not exit_state.standby and not exit_state.holding:
```
Writing this on one line will make the logic clearer.

Reply:
yes! I've updated the code accordingly.